Go 1.15: generated code does not pass go vet

**Describe the bug**
With go 1.15 generated code that does not pass `go test` because of `go vet` warning.

**To Reproduce**
```
$ go version
go version go1.15 darwin/amd64

$ reform --version
v1.4.1

$ cat test.go 
package main


//go:generate reform

//reform:items
type item struct {
	ID string `reform:"id,pk"`
}

$ go generate ./...
$ go test ./...
./test_reform.go:107:10: conversion from int64 to string yields a string of one rune, not a string of digits (did you mean fmt.Sprint(x)?)
```

**Expected behavior**
No error on go vet

**Additional context**
https://github.com/golang/go/issues/32479

--- a/base.go
+++ b/base.go
@@ -81,7 +81,7 @@ type Record interface {
 	// HasPK returns true if record has non-zero primary key set, false otherwise.
 	HasPK() bool
 
-	// SetPK sets record primary key.
+	// SetPK sets record primary key, if possible.
 	//
 	// Deprecated: prefer direct field assignment where possible.
 	SetPK(pk interface{})
@@ -207,14 +207,15 @@ type Dialect interface {
 	DefaultValuesMethod() DefaultValuesMethod
 }
 
-// SetPK sets record's primary key.
+// SetPK sets record's primary key, if possible.
 //
 // Deprecated: prefer direct field assignment where possible.
 func SetPK(r Record, pk interface{}) {
-	t := r.Table()
-	p := r.Pointers()[t.PKColumnIndex()]
-	v := reflect.ValueOf(pk).Int()
-	reflect.ValueOf(p).Elem().SetInt(v)
+	fV := reflect.ValueOf(r.Pointers()[r.Table().PKColumnIndex()]).Elem()
+	pkV := reflect.ValueOf(pk)
+	if t := fV.Type(); t.ConvertibleTo(pkV.Type()) {
+		fV.Set(pkV.Convert(t))
+	}
 }
 
 // check interfaces

--- a/base.go
+++ b/base.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"database/sql"
 	"errors"
+	"reflect"
 )
 
 var (
@@ -81,7 +82,8 @@ type Record interface {
 	HasPK() bool
 
 	// SetPK sets record primary key.
-	// Prefer direct field assignment where possible.
+	//
+	// Deprecated: prefer direct field assignment where possible.
 	SetPK(pk interface{})
 }
 
@@ -203,6 +205,16 @@ type Dialect interface {
 
 	// DefaultValuesMethod returns a method of inserting of row with all default values.
 	DefaultValuesMethod() DefaultValuesMethod
+}
+
+// SetPK sets record's primary key.
+//
+// Deprecated: prefer direct field assignment where possible.
+func SetPK(r Record, pk interface{}) {
+	t := r.Table()
+	p := r.Pointers()[t.PKColumnIndex()]
+	v := reflect.ValueOf(pk).Int()
+	reflect.ValueOf(p).Elem().SetInt(v)
 }
 
 // check interfaces

--- a/base_test.go
+++ b/base_test.go
@@ -2,6 +2,7 @@ package reform_test
 
 import (
 	"database/sql"
+	"flag"
 	"fmt"
 	"log"
 	"os"
@@ -14,6 +15,7 @@ import (
 	_ "github.com/lib/pq"
 	_ "github.com/mattn/go-sqlite3"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/stretchr/testify/suite"
 
@@ -28,12 +30,20 @@ import (
 
 var (
 	// DB is a global connection pool shared by tests and examples.
+	//
 	// Deprecated: do not add new tests using it as using a global pool makes tests more brittle.
 	DB *reform.DB
 )
 
 func TestMain(m *testing.M) {
-	DB = internal.ConnectToTestDB()
+	flag.Parse()
+
+	if testing.Short() {
+		log.Print("Not setting DB in short mode")
+	} else {
+		DB = internal.ConnectToTestDB()
+	}
+
 	os.Exit(m.Run())
 }
 
@@ -301,4 +311,30 @@ func (s *ReformSuite) TestColumns() {
 	s.NoError(err)
 	s.Equal(PersonTable.Columns(), columns)
 	s.NoError(rows.Close())
+}
+
+func TestSetPK(t *testing.T) {
+	t.Parallel()
+
+	var person Person
+	person.SetPK(int32(1))
+	assert.EqualValues(t, 1, person.ID)
+	person.SetPK(int64(2))
+	assert.EqualValues(t, 2, person.ID)
+	person.SetPK(Integer(3))
+	assert.EqualValues(t, 3, person.ID)
+
+	var project Project
+	project.SetPK("baron")
+	assert.EqualValues(t, "baron", project.ID)
+	project.SetPK(1)
+	assert.EqualValues(t, "baron", project.ID)
+
+	var extra Extra
+	extra.SetPK(int32(1))
+	assert.EqualValues(t, 1, extra.ID)
+	extra.SetPK(int64(2))
+	assert.EqualValues(t, 2, extra.ID)
+	extra.SetPK(Integer(3))
+	assert.EqualValues(t, 3, extra.ID)
 }

--- a/base_test.go
+++ b/base_test.go
@@ -313,6 +313,7 @@ func (s *ReformSuite) TestColumns() {
 	s.NoError(rows.Close())
 }
 
+//nolint:staticcheck
 func TestSetPK(t *testing.T) {
 	t.Parallel()
 

--- a/internal/test/models/extra_reform.go
+++ b/internal/test/models/extra_reform.go
@@ -167,13 +167,9 @@ func (s *Extra) HasPK() bool {
 
 // SetPK sets record primary key.
 //
-// Prefer direct field assignment where possible: s.ID = pk.
+// Deprecated: prefer direct field assignment where possible: s.ID = pk.
 func (s *Extra) SetPK(pk interface{}) {
-	if i64, ok := pk.(int64); ok {
-		s.ID = Integer(i64)
-	} else {
-		s.ID = pk.(Integer)
-	}
+	reform.SetPK(s, pk)
 }
 
 // check interfaces
@@ -287,13 +283,9 @@ func (s *notExported) HasPK() bool {
 
 // SetPK sets record primary key.
 //
-// Prefer direct field assignment where possible: s.ID = pk.
+// Deprecated: prefer direct field assignment where possible: s.ID = pk.
 func (s *notExported) SetPK(pk interface{}) {
-	if i64, ok := pk.(int64); ok {
-		s.ID = string(i64)
-	} else {
-		s.ID = pk.(string)
-	}
+	reform.SetPK(s, pk)
 }
 
 // check interfaces

--- a/internal/test/models/extra_reform.go
+++ b/internal/test/models/extra_reform.go
@@ -165,7 +165,7 @@ func (s *Extra) HasPK() bool {
 	return s.ID != ExtraTable.z[ExtraTable.s.PKFieldIndex]
 }
 
-// SetPK sets record primary key.
+// SetPK sets record primary key, if possible.
 //
 // Deprecated: prefer direct field assignment where possible: s.ID = pk.
 func (s *Extra) SetPK(pk interface{}) {
@@ -281,7 +281,7 @@ func (s *notExported) HasPK() bool {
 	return s.ID != notExportedTable.z[notExportedTable.s.PKFieldIndex]
 }
 
-// SetPK sets record primary key.
+// SetPK sets record primary key, if possible.
 //
 // Deprecated: prefer direct field assignment where possible: s.ID = pk.
 func (s *notExported) SetPK(pk interface{}) {

--- a/internal/test/models/good_reform.go
+++ b/internal/test/models/good_reform.go
@@ -135,7 +135,7 @@ func (s *Person) HasPK() bool {
 	return s.ID != PersonTable.z[PersonTable.s.PKFieldIndex]
 }
 
-// SetPK sets record primary key.
+// SetPK sets record primary key, if possible.
 //
 // Deprecated: prefer direct field assignment where possible: s.ID = pk.
 func (s *Person) SetPK(pk interface{}) {
@@ -266,7 +266,7 @@ func (s *Project) HasPK() bool {
 	return s.ID != ProjectTable.z[ProjectTable.s.PKFieldIndex]
 }
 
-// SetPK sets record primary key.
+// SetPK sets record primary key, if possible.
 //
 // Deprecated: prefer direct field assignment where possible: s.ID = pk.
 func (s *Project) SetPK(pk interface{}) {
@@ -462,7 +462,7 @@ func (s *IDOnly) HasPK() bool {
 	return s.ID != IDOnlyTable.z[IDOnlyTable.s.PKFieldIndex]
 }
 
-// SetPK sets record primary key.
+// SetPK sets record primary key, if possible.
 //
 // Deprecated: prefer direct field assignment where possible: s.ID = pk.
 func (s *IDOnly) SetPK(pk interface{}) {
@@ -583,7 +583,7 @@ func (s *Constraints) HasPK() bool {
 	return s.ID != ConstraintsTable.z[ConstraintsTable.s.PKFieldIndex]
 }
 
-// SetPK sets record primary key.
+// SetPK sets record primary key, if possible.
 //
 // Deprecated: prefer direct field assignment where possible: s.ID = pk.
 func (s *Constraints) SetPK(pk interface{}) {
@@ -705,7 +705,7 @@ func (s *LegacyPerson) HasPK() bool {
 	return s.ID != LegacyPersonTable.z[LegacyPersonTable.s.PKFieldIndex]
 }
 
-// SetPK sets record primary key.
+// SetPK sets record primary key, if possible.
 //
 // Deprecated: prefer direct field assignment where possible: s.ID = pk.
 func (s *LegacyPerson) SetPK(pk interface{}) {

--- a/internal/test/models/good_reform.go
+++ b/internal/test/models/good_reform.go
@@ -137,13 +137,9 @@ func (s *Person) HasPK() bool {
 
 // SetPK sets record primary key.
 //
-// Prefer direct field assignment where possible: s.ID = pk.
+// Deprecated: prefer direct field assignment where possible: s.ID = pk.
 func (s *Person) SetPK(pk interface{}) {
-	if i64, ok := pk.(int64); ok {
-		s.ID = int32(i64)
-	} else {
-		s.ID = pk.(int32)
-	}
+	reform.SetPK(s, pk)
 }
 
 // check interfaces
@@ -272,13 +268,9 @@ func (s *Project) HasPK() bool {
 
 // SetPK sets record primary key.
 //
-// Prefer direct field assignment where possible: s.ID = pk.
+// Deprecated: prefer direct field assignment where possible: s.ID = pk.
 func (s *Project) SetPK(pk interface{}) {
-	if i64, ok := pk.(int64); ok {
-		s.ID = string(i64)
-	} else {
-		s.ID = pk.(string)
-	}
+	reform.SetPK(s, pk)
 }
 
 // check interfaces
@@ -472,13 +464,9 @@ func (s *IDOnly) HasPK() bool {
 
 // SetPK sets record primary key.
 //
-// Prefer direct field assignment where possible: s.ID = pk.
+// Deprecated: prefer direct field assignment where possible: s.ID = pk.
 func (s *IDOnly) SetPK(pk interface{}) {
-	if i64, ok := pk.(int64); ok {
-		s.ID = int32(i64)
-	} else {
-		s.ID = pk.(int32)
-	}
+	reform.SetPK(s, pk)
 }
 
 // check interfaces
@@ -597,13 +585,9 @@ func (s *Constraints) HasPK() bool {
 
 // SetPK sets record primary key.
 //
-// Prefer direct field assignment where possible: s.ID = pk.
+// Deprecated: prefer direct field assignment where possible: s.ID = pk.
 func (s *Constraints) SetPK(pk interface{}) {
-	if i64, ok := pk.(int64); ok {
-		s.ID = string(i64)
-	} else {
-		s.ID = pk.(string)
-	}
+	reform.SetPK(s, pk)
 }
 
 // check interfaces
@@ -723,13 +707,9 @@ func (s *LegacyPerson) HasPK() bool {
 
 // SetPK sets record primary key.
 //
-// Prefer direct field assignment where possible: s.ID = pk.
+// Deprecated: prefer direct field assignment where possible: s.ID = pk.
 func (s *LegacyPerson) SetPK(pk interface{}) {
-	if i64, ok := pk.(int64); ok {
-		s.ID = int32(i64)
-	} else {
-		s.ID = pk.(int32)
-	}
+	reform.SetPK(s, pk)
 }
 
 // check interfaces

--- a/querier_commands.go
+++ b/querier_commands.go
@@ -94,7 +94,9 @@ func (q *Querier) insert(str Struct, columns []string, values []interface{}) err
 			if err != nil {
 				return err
 			}
-			record.SetPK(id) // TODO optimize
+
+			// TODO optimize to avoid using reflection
+			record.SetPK(id) //nolint:staticcheck
 		}
 		return nil
 

--- a/querier_commands.go
+++ b/querier_commands.go
@@ -94,7 +94,7 @@ func (q *Querier) insert(str Struct, columns []string, values []interface{}) err
 			if err != nil {
 				return err
 			}
-			record.SetPK(id)
+			record.SetPK(id) // TODO optimize
 		}
 		return nil
 

--- a/reform-db/base_test.go
+++ b/reform-db/base_test.go
@@ -19,6 +19,10 @@ func TestReformDBSuite(t *testing.T) {
 }
 
 func (s *ReformDBSuite) SetupSuite() {
+	if testing.Short() {
+		s.T().Skip("skipping in short mode")
+	}
+
 	logger = internal.NewLogger("reform-db-test: ", true)
 
 	s.db = internal.ConnectToTestDB()

--- a/reform/template.go
+++ b/reform/template.go
@@ -127,13 +127,9 @@ func (s *{{ .Type }}) HasPK() bool {
 
 // SetPK sets record primary key.
 //
-// Prefer direct field assignment where possible: s.{{ .PKField.Name }} = pk.
+// Deprecated: prefer direct field assignment where possible: s.{{ .PKField.Name }} = pk.
 func (s *{{ .Type }}) SetPK(pk interface{}) {
-	if i64, ok := pk.(int64); ok {
-		s.{{ .PKField.Name }} = {{ .PKField.Type }}(i64)
-	} else {
-		s.{{ .PKField.Name }} = pk.({{ .PKField.Type }})
-	}
+	reform.SetPK(s, pk)
 }
 
 {{- end }}

--- a/reform/template.go
+++ b/reform/template.go
@@ -125,7 +125,7 @@ func (s *{{ .Type }}) HasPK() bool {
 	return s.{{ .PKField.Name }} != {{ .TableVar }}.z[{{ .TableVar }}.s.PKFieldIndex]
 }
 
-// SetPK sets record primary key.
+// SetPK sets record primary key, if possible.
 //
 // Deprecated: prefer direct field assignment where possible: s.{{ .PKField.Name }} = pk.
 func (s *{{ .Type }}) SetPK(pk interface{}) {

